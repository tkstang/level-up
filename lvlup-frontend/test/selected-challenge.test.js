import SelectedChallenge from '../src/components/Admin/challenges/admin-challenges/individual-submission';
import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import lvlupApp from '../src/reducers/index';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

test('SelectedChallenge should render', () => {
  const store = mockStore({ lvlupApp });
  const challenge = shallow(
    <SelectedChallenge selectedChallenge={{ id: 1, challenge: { name: 'Feed the Hamid' }, category: { category: 'Life' }, student: 'Thomas Stang' }} adminLoginInfo={{ cohorts: ['g42', 'g52'] }} store={store} />,
  );
  expect(shallowToJson(challenge)).toMatchSnapshot();
});
