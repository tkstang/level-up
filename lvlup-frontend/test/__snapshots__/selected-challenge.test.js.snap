exports[`test SelectedChallenge should render 1`] = `
<div
  className="lvl-table">
  <Container>
    <h1
      className="header">
      Challenge Submission Review: Feed the Hamid
    </h1>
    <div>
      <h3>
        undefined, ,
      </h3>
      <Table
        as="table"
        celled={true}
        selectable={true}>
        <TableHeader
          as="thead">
          <TableRow
            as="tr"
            cellAs="td">
            <TableHeaderCell
              as="th"
              className="table-head"
              colSpan="5"
              id="challenge-info"
              textAlign="center">
              Challenge Information
            </TableHeaderCell>
          </TableRow>
          <TableRow
            as="tr"
            cellAs="td">
            <TableHeaderCell
              as="th"
              className="thead-secondary"
              textAlign="center">
              Title
            </TableHeaderCell>
            <TableHeaderCell
              as="th"
              className="thead-secondary"
              textAlign="center">
              Category
            </TableHeaderCell>
            <TableHeaderCell
              as="th"
              className="thead-secondary"
              textAlign="center">
              Description
            </TableHeaderCell>
            <TableHeaderCell
              as="th"
              className="thead-secondary"
              textAlign="center">
              Requirement
            </TableHeaderCell>
            <TableHeaderCell
              as="th"
              className="thead-secondary"
              textAlign="center">
              Points
            </TableHeaderCell>
          </TableRow>
        </TableHeader>
        <TableBody
          as="tbody">
          <TableRow
            as="tr"
            cellAs="td">
            <TableCell
              as="td"
              textAlign="center">
              Feed the Hamid
            </TableCell>
            <TableCell
              as="td"
              textAlign="center">
              Life
            </TableCell>
            <TableCell
              as="td"
              textAlign="center" />
            <TableCell
              as="td"
              textAlign="center">
              No requirements!
            </TableCell>
            <TableCell
              as="td"
              textAlign="center" />
          </TableRow>
        </TableBody>
      </Table>
      <Table
        as="table"
        celled={true}
        selectable={true}>
        <TableHeader
          as="thead">
          <TableRow
            as="tr"
            cellAs="td">
            <TableHeaderCell
              as="th"
              className="table-head"
              colSpan="5"
              id="submission-info"
              textAlign="center">
              Submission Information
            </TableHeaderCell>
          </TableRow>
          <TableRow
            as="tr"
            cellAs="td">
            <TableHeaderCell
              as="th"
              className="thead-secondary"
              textAlign="center">
              Submission Message
            </TableHeaderCell>
            <TableHeaderCell
              as="th"
              className="thead-secondary"
              textAlign="center">
              Attachment 1
            </TableHeaderCell>
            <TableHeaderCell
              as="th"
              className="thead-secondary"
              textAlign="center">
              Attachment 2
            </TableHeaderCell>
            <TableHeaderCell
              as="th"
              className="thead-secondary"
              textAlign="center">
              Attachment 3
            </TableHeaderCell>
            <TableHeaderCell
              as="th"
              className="thead-secondary"
              textAlign="center">
              Images
            </TableHeaderCell>
          </TableRow>
        </TableHeader>
        <TableBody
          as="tbody">
          <TableRow
            as="tr"
            cellAs="td">
            <TableCell
              as="td" />
            <TableCell
              as="td">
              No Attachment Available
            </TableCell>
            <TableCell
              as="td">
              No Attachment Available
            </TableCell>
            <TableCell
              as="td">
              No Attachment Available
            </TableCell>
            <TableCell
              as="td">
              No phots submitted
            </TableCell>
          </TableRow>
        </TableBody>
      </Table>
      <Connect(ReduxForm) />
    </div>
  </Container>
</div>
`;
